assets/js/utils/style-attributes-utils.ts

```diff
@@ -0,0 +1,123 @@
+/* eslint-disable @wordpress/no-unsafe-wp-apis */
+/**
+ * External dependencies
+ */
+import { __experimentalUseColorProps } from '@wordpress/block-editor';
+
+/**
+ * Internal dependencies
+ */
+import { isFeaturePluginBuild } from '../settings/blocks/feature-flags';
+import { isString, isObject } from '../types/type-guards';
+
+type WithClass = {
+	className: string;
+};
+
+type WithStyle = {
+	style: Record< string, unknown >;
+};
+
+const parseStyle = ( style: unknown ): Record< string, unknown > => {
+	if ( isString( style ) ) {
+		return JSON.parse( style ) || {};
+	}
+
+	if ( isObject( style ) ) {
+		return style;
+	}
+
+	return {};
+};
+
+const getSpacingStyleInline = (
+	key: string,
+	values: {
+		top: string | null;
+		right: string | null;
+		bottom: string | null;
+		left: string | null;
+	}
+) => {
+	return {
+		...( isString( values.top ) && { [ `${ key }Top` ]: values.top } ),
+		...( isString( values.right ) && {
+			[ `${ key }Right` ]: values.right,
+		} ),
+		...( isString( values.bottom ) && {
+			[ `${ key }Bottom` ]: values.bottom,
+		} ),
+		...( isString( values.left ) && { [ `${ key }Left` ]: values.left } ),
+	};
+};
+
+const parseSpacingStyle = (
+	spacing: Record< string, unknown >
+): Record< string, unknown > => {
+	const keys = [ 'margin' ];
+
+	const getValueOrDefault = ( value: unknown ) => {
+		return isString( value ) && value.length > 0 ? value : null;
+	};
+
+	return Object.keys( spacing ).reduce( ( acc, key ) => {
+		const spacingProperty = isObject( spacing[ key ] )
+			? ( spacing[ key ] as Record< string, unknown > )
+			: {};
+
+		if ( keys.includes( key ) ) {
+			return {
+				...acc,
+				...getSpacingStyleInline( key, {
+					top: getValueOrDefault( spacingProperty.top ),
+					right: getValueOrDefault( spacingProperty.right ),
+					bottom: getValueOrDefault( spacingProperty.bottom ),
+					left: getValueOrDefault( spacingProperty.left ),
+				} ),
```

Comment:
> I think we can pass spacingProperty as the second argument then handle sides there instead. It makes the callback look cleaner.

It can be an idea, but to be honest I prefer to create a more strict typed code. 

![image](https://user-images.githubusercontent.com/4463174/148389640-401671ed-d157-4b08-bdb3-91f1ed8e87fe.png)


With this code (so without type), we lose some information about how the function `getSpacingStyleInline` works.
I mean without an object with `top`, `right`, `bottom`, and `left` key, it doesn't make sense invoke getSpacingStyleInline` function. With the type `Record<string, unknown>` we lose this information and the developer is forced to read the whole function to understand what kind of data to pass.

But this is just my opinion, if you think that it is better to change it, I will do it :D

Reply:
@gigitux With your example, I don't see the need for `getSpacingStyleInline`, we can add the inline style directly like this. Note that we can use `undefined` properties just fine because they will get omitted during render.

```
		if ( keys.includes( key ) ) {
			return {
				...acc,
				[ `${ key }Left` ]: spacingProperty.left,
				[ `${ key }Right` ]: spacingProperty.right,
				[ `${ key }Top` ]: spacingProperty.top,
				[ `${ key }Bottom` ]: spacingProperty.bottom,
			};
		}
```